e2e/framework/test_context.go

```diff
@@ -375,7 +351,22 @@ func newContextConfig(configFile string) error {
 		return trace.Wrap(err)
 	}
 
-	return trace.Wrap(yaml.Unmarshal(configBytes, &TestContext))
+	err = yaml.Unmarshal(configBytes, &TestContext)
+	if err != nil {
+		return trace.Wrap(err, "Error parsing config file")
+	}
+
+	err = validator.New().Struct(TestContext)
+	if err == nil {
+		return nil
+	}
+
+	log.Errorf("Configuration file %s has errors", configFile)
+	for _, fieldError := range err.(validator.ValidationErrors) {
+		log.Errorf("   Field %s=%v fails rule %s", fieldError.Field(), fieldError.Value(), fieldError.Tag())
+	}
+
+	return trace.Errorf("Configuration file fails validation test")
```

Comment:
Lower-case first letter of error messages, for better embedding: https://github.com/golang/go/wiki/CodeReviewComments#error-strings (above too).